Mathlib/Data/Fintype/Pi.lean

```diff
@@ -133,3 +135,43 @@ lemma Fin.snoc_mem_piFinset_snoc_iff {n : ℕ} {α : Fin (n + 1) → Type*}
     (Sᵢ : (i : Fin n) → Finset (α i.castSucc)) (Sₙ : Finset (α <| .last n)) :
     Fin.snoc xs x ∈ Fintype.piFinset (Fin.snoc Sᵢ Sₙ) ↔ xs ∈ Fintype.piFinset Sᵢ ∧ x ∈ Sₙ := by
   simp_rw [Fin.mem_piFinset_succ_iff', init_snoc, snoc_last]
+
+lemma Finset.map_piFinSuccAboveEquiv_filter_piFinset_succAbove {n : ℕ} (k : Fin (n + 1))
```

Comment:
Those names are really hard to parse. Can you
* Refer to the unbundled function rather than the bundled `Equiv`. The fact that it's bundled is mostly an implementation detail stemming from using `Finset.map`.
* Use the correct `Equiv` rather than the general `Equiv.piFinSuccAboveEquiv`. We have `Equiv.piFinSucc`, but not `Equiv.piFinCastSucc`. So you first need a PR to add `Equiv.piFinCastSucc`, and it would be great if you also renamed `Equiv.piFinSuccAboveEquiv` to `Equiv.piFinSuccAbove` to match.

Reply:
Ok the PRs renaming to remove the duplicated `Equiv` in the name and adding `Equiv.piFinCastSucc` are merged.

For the first point, I am looking at the definition of `Equiv.piFinSuccAbove` and not seeing `Finset.map`. I guess you are saying though saying that another def should be introduced for the function form of `Equiv.piFinSuccAbove` and `Equiv.piFinSuccAbove` should then be redefined as that?

Reply:
The inverse function is "insertNth". "extractNth" perhaps?

Reply:
#10545

Reply:
That's a reasonable interpretation of my comment. `extractNth` seems like a reasonable name.

Reply:
Ok after doing all that, it seems like it's impossible to substitute this in because map actually takes an embedding, not a function. I am not sure if you wanted something else from me here?